tools/fastq_screen.wdl

```diff
@@ -50,7 +50,7 @@ task fastq_screen {
     Float db_size = size(db, "GiB")
     Int disk_size = ceil(db_size * 2)
 
-    String output_basename = basename(read1, "R1.fastq")
+    String sample_basename = basename(read1, "_R1.fastq")
```

Comment:
Any assumptions about sample naming should be bubbled up to workflow options so that others can change it based on their format.

Reply:
`_R1`/`_R2` are currently hardcoded suffixes in `picard.bam_to_fastq`, and I think any way to optionally set the suffixes trimmed here is going to be messy and unused. I don't think a use case for calling this without `picard.bam_to_fastq` first exists.

The best way I can think of to handle this with mutable suffixes would be to allow a workflow option for changing `_R`, and then applying `1/2` after the user supplied suffix (Otherwise there would need to be an option for a read 1 suffix and an option for a read 2 suffix). That option would also need to be passed to `picard.bam_to_fastq`, and potentially other tools making this assumption (I haven't checked if there are any).

Personally I think that's introducing an unnecessary complication. It will make the code messier, and introduce workflow/tool options for the user to review that will likely never be used. Are there any use cases you can think of where this would be useful?

Reply:
You could make sample name an optional parameter. Then do something like `select_first(sample_name, inferred_name)`.